src/coreclr/vm/eventing/eventpipe/ep-rt-coreclr.h

```diff
@@ -1569,7 +1569,34 @@ ep_rt_config_value_get_output_path (void)
 {
 	STATIC_CONTRACT_NOTHROW;
 	CLRConfigStringHolder value(CLRConfig::GetConfigValue (CLRConfig::INTERNAL_EventPipeOutputPath));
-	return ep_rt_utf16_to_utf8_string (reinterpret_cast<ep_char16_t *>(value.GetValue ()), -1);
+	ep_char8_t *outputPath = ep_rt_utf16_to_utf8_string (reinterpret_cast<ep_char16_t *>(value.GetValue ()), -1);
+	if (outputPath != NULL)
+	{
+		while (true)
+		{
+			const char* pidSearchString = "{pid}";
+			ep_char8_t* outputPathPid = strstr(outputPath, pidSearchString);
+			if (outputPathPid != NULL)
+			{
+				size_t newBufSize = strlen(outputPath) + 12; // 12 is 1 for the null-terminator + 11 for the largest possible 4 byte integer converted to a string
```

Comment:
If I recall, macOS uses 64-bit PIDs.  Does `GetCurrentProcessId` account for that or does it always truncate to a 32-bit value?

Reply:
@janvorli Do I need to worry about 64bit pids? I was assuming that the PAL would insulate me here, but if some supported OS might actually use a 64bit pid one could get in trouble. Do you have any data here?